### ISSUE TITLE
Compiler errors not being reported (unless they are syntax errors)

### ISSUE THREAD

**Issue by AaronM04** [rust-analyzer/rust-analyzer#5918]
**Steps to Reproduce**
1. Clone https://github.com/conwayste/conwayste
2. Checkout commit e07f87b
3. Open up `conwayste/src/client.rs` with neovim running LanguageClient-neovim
4. Go to line 283, and insert the line `let () = net_worker;` (such that this is the new line 283)
5. Click Save

**Expected Result**
An error about type mismatches showing the type of `net_worker`

**Actual Result**
No errors shown.

**Version Info**
rustc 1.45.2
rust-analyzer 0b2b9a5

**Log**
[rust-analyzer-bug.log](https://github.com/rust-analyzer/rust-analyzer/files/5147313/rust-analyzer-bug.log)

**Comment by staninprague** [comment]
Not sure this would be exactly the same root cause. This happens to me as well, but only when I point to anything external in one of the Cargo.toml in the workspace, like:
```
[dependencies.a2]
 path="../../a2"
```
It works for some time. And then it stops running cargo check at some moment. Restarting vscode would not help. Re-installing rust-analyzer helps for some minutes of work again.

**Comment by AaronM04** [comment]
Interesting observation. If you check our repo you'll see we are also doing this: https://github.com/conwayste/conwayste/blob/master/conwayste/Cargo.toml

**Comment by lnicola** [comment]
Triage: I can't reproduce this because `cargo check` fails with a weird linker error:

```
    Checking winit v0.19.5
error: linking with `clang` failed: exit code: 1
  |
  = note: "clang" "-Wl,--as-needed" "-Wl,-z,noexecstack" "-m64" "-Wl,--eh-frame-hdr" "-L" "/home/me/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "/home/me/conwayste/target/debug/build/dissect-netwayste-648cfd707c4b060b/build_script_build-648cfd707c4b060b.13kedvlsesdmyrn2.rcgu.o" "/home/me/conwayste/target/debug/build/dissect-netwayste-648cfd707c4b060b/build_script_build-648cfd707c4b060b.1rw7am10i2xacag9.rcgu.o" "/home/me/conwayste/target/debug/build/dissect-netwayste-648cfd707c4b060b/build_script_build-648cfd707c4b060b.2ewpp7u4fdo2z7wf.rcgu.o" "/home/me/conwayste/target/debug/build/dissect-netwayste-648cfd707c4b060b/build_script_build-648cfd707c4b060b.2vnu8arzaqsrtel5.rcgu.o" "/home/me/conwayste/target/debug/build/dissect-netwayste-648cfd707c4b060b/build_script_build-648cfd707c4b060b.37kmzq0s29y0nl0q.rcgu.o" "/home/me/conwayste/target/debug/build/dissect-netwayste-648cfd707c4b060b/build_script_build-648cfd707c4b060b.3idybq3wwguixdxw.rcgu.o" "/home/me/conwayste/target/debug/build/dissect-netwayste-648cfd707c4b060b/build_script_build-648cfd707c4b060b.3vh6267wmiaw9ch4.rcgu.o" "/home/me/conwayste/target/debug/build/dissect-netwayste-648cfd707c4b060b/build_script_build-648cfd707c4b060b.3vhq4d6fwjy6jli0.rcgu.o" "/home/me/conwayste/target/debug/build/dissect-netwayste-648cfd707c4b060b/build_script_build-648cfd707c4b060b.3z0mroztar7xr6cd.rcgu.o" "/home/me/conwayste/target/debug/build/dissect-netwayste-648cfd707c4b060b/build_script_build-648cfd707c4b060b.4045bdam7x43jejm.rcgu.o" "/home/me/conwayste/target/debug/build/dissect-netwayste-648cfd707c4b060b/build_script_build-648cfd707c4b060b.42065utw7d4u9uj7.rcgu.o" "/home/me/conwayste/target/debug/build/dissect-netwayste-648cfd707c4b060b/build_script_build-648cfd707c4b060b.42y9f61z54omk4re.rcgu.o" "/home/me/conwayste/target/debug/build/dissect-netwayste-648cfd707c4b060b/build_script_build-648cfd707c4b060b.4dpt4r2s8bs3clsl.rcgu.o" "/home/me/conwayste/target/debug/build/dissect-netwayste-648cfd707c4b060b/build_script_build-648cfd707c4b060b.4est0yji334iwbzy.rcgu.o" "/home/me/conwayste/target/debug/build/dissect-netwayste-648cfd707c4b060b/build_script_build-648cfd707c4b060b.4fjiughrlry6zl6q.rcgu.o" "/home/me/conwayste/target/debug/build/dissect-netwayste-648cfd707c4b060b/build_script_build-648cfd707c4b060b.4fsmg21blreue2wh.rcgu.o" "/home/me/conwayste/target/debug/build/dissect-netwayste-648cfd707c4b060b/build_script_build-648cfd707c4b060b.4rr82ong81ga4kdx.rcgu.o" "/home/me/conwayste/target/debug/build/dissect-netwayste-648cfd707c4b060b/build_script_build-648cfd707c4b060b.4vht20nuw1pp3atg.rcgu.o" "/home/me/conwayste/target/debug/build/dissect-netwayste-648cfd707c4b060b/build_script_build-648cfd707c4b060b.52hl0bkkx1c7ahf5.rcgu.o" "/home/me/conwayste/target/debug/build/dissect-netwayste-648cfd707c4b060b/build_script_build-648cfd707c4b060b.54vuwkwl9kc4y44z.rcgu.o" "/home/me/conwayste/target/debug/build/dissect-netwayste-648cfd707c4b060b/build_script_build-648cfd707c4b060b.59wpmmih8y73u4ql.rcgu.o" "/home/me/conwayste/target/debug/build/dissect-netwayste-648cfd707c4b060b/build_script_build-648cfd707c4b060b.9itj0p9987njnuq.rcgu.o" "/home/me/conwayste/target/debug/build/dissect-netwayste-648cfd707c4b060b/build_script_build-648cfd707c4b060b.aemw3ids50rt9g9.rcgu.o" "/home/me/conwayste/target/debug/build/dissect-netwayste-648cfd707c4b060b/build_script_build-648cfd707c4b060b.lvcdl0ealcimlen.rcgu.o" "/home/me/conwayste/target/debug/build/dissect-netwayste-648cfd707c4b060b/build_script_build-648cfd707c4b060b.qo5jxi90t7htuya.rcgu.o" "-o" "/home/me/conwayste/target/debug/build/dissect-netwayste-648cfd707c4b060b/build_script_build-648cfd707c4b060b" "/home/me/conwayste/target/debug/build/dissect-netwayste-648cfd707c4b060b/build_script_build-648cfd707c4b060b.4b1xz45v40985vnp.rcgu.o" "-Wl,--gc-sections" "-pie" "-Wl,-zrelro" "-Wl,-znow" "-nodefaultlibs" "-L" "/home/me/conwayste/target/debug/deps" "-L" "/home/me/conwayste/target/debug/build/libloading-8807affcfcf64941/out" "-L" "/home/me/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-Wl,-Bstatic" "/home/me/conwayste/target/debug/deps/libpkg_config-e6be48dfd8446a62.rlib" "/home/me/conwayste/target/debug/deps/libbindgen-1db9183bc6de7cb7.rlib" "/home/me/conwayste/target/debug/deps/liblog-4bf1047162af990b.rlib" "/home/me/conwayste/target/debug/deps/libwhich-3f0cf2a884436b94.rlib" "/home/me/conwayste/target/debug/deps/libshlex-4a4c2f8255e6015b.rlib" "/home/me/conwayste/target/debug/deps/libregex-9fca807dbf6037c4.rlib" "/home/me/conwayste/target/debug/deps/libthread_local-b1b857567b38aff1.rlib" "/home/me/conwayste/target/debug/deps/libregex_syntax-f58e76f38029c5ae.rlib" "/home/me/conwayste/target/debug/deps/libaho_corasick-795ec644f577bef4.rlib" "/home/me/conwayste/target/debug/deps/libquote-21d96aae44871e50.rlib" "/home/me/conwayste/target/debug/deps/libproc_macro2-6ca644c400011813.rlib" "/home/me/conwayste/target/debug/deps/libunicode_xid-083414e3434d8e82.rlib" "/home/me/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libproc_macro-eba4ba6c63f96a02.rlib" "/home/me/conwayste/target/debug/deps/libpeeking_take_while-209dce6f2d242089.rlib" "/home/me/conwayste/target/debug/deps/liblazy_static-cd2401a06b1676f6.rlib" "/home/me/conwayste/target/debug/deps/librustc_hash-d8014d7e08dbd39d.rlib" "/home/me/conwayste/target/debug/deps/liblazycell-13e71b09752696d8.rlib" "/home/me/conwayste/target/debug/deps/libclang_sys-9958a00734c6b066.rlib" "/home/me/conwayste/target/debug/deps/liblibloading-11977ad0b39072b1.rlib" "/home/me/conwayste/target/debug/deps/liblibc-6ba8b754d6fb20a6.rlib" "/home/me/conwayste/target/debug/deps/libglob-c57da5286c7fae6e.rlib" "/home/me/conwayste/target/debug/deps/libcfg_if-4c6be1962c556cf0.rlib" "/home/me/conwayste/target/debug/deps/libcexpr-498c25af64543d0e.rlib" "/home/me/conwayste/target/debug/deps/libnom-ad1c863c910eb478.rlib" "/home/me/conwayste/target/debug/deps/libmemchr-bb23920ec89494b5.rlib" "/home/me/conwayste/target/debug/deps/libbitflags-df474a8ca45884a8.rlib" "-Wl,--start-group" "/home/me/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-25c6acf8063a3802.rlib" "/home/me/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libpanic_unwind-539f13c9442f1597.rlib" "/home/me/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libminiz_oxide-37db28e905edb56b.rlib" "/home/me/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libadler-9ef2480568df55af.rlib" "/home/me/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libobject-1e0f0992cdbecd66.rlib" "/home/me/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libaddr2line-6c8e02b8fedc1e5f.rlib" "/home/me/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libgimli-807e5ad203594490.rlib" "/home/me/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_demangle-083fce1bea11612a.rlib" "/home/me/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libhashbrown-1af568081add9042.rlib" "/home/me/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_std_workspace_alloc-1395b54a3b3f45bf.rlib" "/home/me/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libunwind-caba820045f178d5.rlib" "/home/me/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcfg_if-512eb53291f6de7e.rlib" "/home/me/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liblibc-5efacc5025f9f3d8.rlib" "/home/me/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liballoc-9c4002b5f79ba0e1.rlib" "/home/me/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_std_workspace_core-90996f4879673567.rlib" "/home/me/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcore-9ea09a899c3eda46.rlib" "-Wl,--end-group" "/home/me/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-ef2408da76957905.rlib" "-Wl,-Bdynamic" "-ldl" "-lutil" "-ldl" "-lutil" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc"
  = note: /usr/bin/ld: /home/me/conwayste/target/debug/build/dissect-netwayste-648cfd707c4b060b/build_script_build-648cfd707c4b060b.3z0mroztar7xr6cd.rcgu.o: in function `build_script_build::main':
          /home/me/conwayste/dissect-netwayste/build.rs:28: undefined reference to `bindgen::Builder::generate'
          /usr/bin/ld: /home/me/conwayste/dissect-netwayste/build.rs:31: undefined reference to `core::result::Result<T,E>::expect'
          /usr/bin/ld: /home/me/conwayste/dissect-netwayste/build.rs:34: undefined reference to `core::ptr::drop_in_place'
          /usr/bin/ld: /home/me/conwayste/dissect-netwayste/build.rs:34: undefined reference to `core::ptr::drop_in_place'
          /usr/bin/ld: /home/me/conwayste/dissect-netwayste/build.rs:34: undefined reference to `core::ptr::drop_in_place'
          /usr/bin/ld: /home/me/conwayste/target/debug/build/dissect-netwayste-648cfd707c4b060b/build_script_build-648cfd707c4b060b.42065utw7d4u9uj7.rcgu.o: in function `<bindgen::Builder as core::default::Default>::default':
          /home/me/.cargo/registry/src/github.com-1ecc6299db9ec823/bindgen-0.53.1/src/lib.rs:204: undefined reference to `<bindgen::BindgenOptions as core::default::Default>::default'
          /usr/bin/ld: /home/me/.cargo/registry/src/github.com-1ecc6299db9ec823/bindgen-0.53.1/src/lib.rs:202: undefined reference to `core::ptr::drop_in_place'
          /usr/bin/ld: /home/me/conwayste/target/debug/build/dissect-netwayste-648cfd707c4b060b/build_script_build-648cfd707c4b060b.59wpmmih8y73u4ql.rcgu.o: in function `bindgen::Builder::header':
          /home/me/.cargo/registry/src/github.com-1ecc6299db9ec823/bindgen-0.53.1/src/lib.rs:683: undefined reference to `core::ptr::drop_in_place'
          /usr/bin/ld: /home/me/conwayste/target/debug/build/dissect-netwayste-648cfd707c4b060b/build_script_build-648cfd707c4b060b.59wpmmih8y73u4ql.rcgu.o: in function `bindgen::Builder::clang_arg':
          /home/me/.cargo/registry/src/github.com-1ecc6299db9ec823/bindgen-0.53.1/src/lib.rs:1059: undefined reference to `core::ptr::drop_in_place'
          /usr/bin/ld: /home/me/conwayste/target/debug/build/dissect-netwayste-648cfd707c4b060b/build_script_build-648cfd707c4b060b.59wpmmih8y73u4ql.rcgu.o: in function `bindgen::Builder::clang_arg':
          /home/me/.cargo/registry/src/github.com-1ecc6299db9ec823/bindgen-0.53.1/src/lib.rs:1059: undefined reference to `core::ptr::drop_in_place'
          /usr/bin/ld: /home/me/conwayste/target/debug/build/dissect-netwayste-648cfd707c4b060b/build_script_build-648cfd707c4b060b.59wpmmih8y73u4ql.rcgu.o: in function `bindgen::Bindings::write_to_file':
          /home/me/.cargo/registry/src/github.com-1ecc6299db9ec823/bindgen-0.53.1/src/lib.rs:2134: undefined reference to `<core::result::Result<T,E> as core::ops::try::Try>::from_error'
          /usr/bin/ld: /home/me/.cargo/registry/src/github.com-1ecc6299db9ec823/bindgen-0.53.1/src/lib.rs:2139: undefined reference to `bindgen::Bindings::write'
          /usr/bin/ld: /home/me/.cargo/registry/src/github.com-1ecc6299db9ec823/bindgen-0.53.1/src/lib.rs:2139: undefined reference to `<core::result::Result<T,E> as core::ops::try::Try>::from_error'
          /usr/bin/ld: /home/me/conwayste/target/debug/deps/libbindgen-1db9183bc6de7cb7.rlib(bindgen-1db9183bc6de7cb7.bindgen.3307kqji-cgu.11.rcgu.o): in function `core::slice::index::<impl core::ops::index::IndexMut<I> for [T]>::index_mut':
          /home/me/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/index.rs:26: undefined reference to `<core::ops::range::RangeFrom<usize> as core::slice::index::SliceIndex<[T]>>::index_mut'
          clang-11: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by AaronM04** [comment]
@lnicola I don't know what the problem is, but `dissect-netwayste` uses bindgen to create a wireshark plugin. It isn't core functionality of this project, so if you can find a way to exclude it from `cargo check` you should be able to reproduce the bug.

**Comment by lnicola** [comment]
@AaronM04 haha, nice, I removed it from `Cargo.toml`. Unfortunately, it's working for me:

![image](https://user-images.githubusercontent.com/308347/102796465-02347300-43b7-11eb-87cf-55fe43cc096e.png)

![image](https://user-images.githubusercontent.com/308347/102796496-0cef0800-43b7-11eb-8c46-0541df0029e9.png)

**Comment by bjorn3** [comment]
Could the linker error you saw @lnicola be the reason that no errors are reported for @AaronM04?

**Comment by lnicola** [comment]
Maybe, but it seems unlikely.

**Comment by AaronM04** [comment]
Is this with the latest rust-analyzer or the version I mentioned?

On December 21, 2020 8:05:27 AM PST, "Laurențiu Nicola" <notifications@github.com> wrote:
>@AaronM04 haha, nice, I removed it from `Cargo.toml`. Unfortunately,
>it's working for me:
>
>![image](https://user-images.githubusercontent.com/308347/102796465-02347300-43b7-11eb-87cf-55fe43cc096e.png)
>
>![image](https://user-images.githubusercontent.com/308347/102796496-0cef0800-43b7-11eb-8c46-0541df0029e9.png)
>
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/rust-analyzer/rust-analyzer/issues/5918#issuecomment-749050593

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by lnicola** [comment]
It also works for me on your version (and nvim/LanguageClient-neovim):

![image](https://user-images.githubusercontent.com/308347/102802390-3ad84a80-43bf-11eb-8492-ef70c3d0b03c.png)

**Comment by lnicola** [comment]
@AaronM04 are you still seeing this?

(Marking `S-actionable`, if you look at this in the future and the OP hasn't answered, ping me so we can close it.)

**Comment by AaronM04** [comment]
@lnicola yes, I am still seeing this.

```
$ rustc --version
rustc 1.48.0
$ uname -a
OpenBSD <redacted hostname> 6.8 GENERIC.MP#243 amd64
```
Rust-analyzer commit `ac30710` from 2020-11-29

Tested with both coc.nvim on vim and with languageClient-neovim on nvim.

**Comment by AaronM04** [comment]
I upgraded rust-analyzer and it is still happening (tested with coc.nvim on vim):

```
$ rust-analyzer --version
rust-analyzer 3ab8d7a
```
This commit is from 2021-01-24

**Comment by lnicola** [comment]
Hmm, is there an OpenBSD live image? I have absolutely no experience with it. I wonder if it might be something platform-specific.

**Comment by AaronM04** [comment]
I don't believe so, but it is pretty quick to install in VirtualBox. `sysupgrade -s` as root installs the latest snapshot, which will then allow you to install a recent Rust version with `pkg_add rust`.

I can assist with editor setup if/when you need assistance.

On January 28, 2021 8:41:32 AM PST, "Laurențiu Nicola" <notifications@github.com> wrote:
>Hmm, is there an OpenBSD live image? I have absolutely no experience
>with it. I wonder if it might be something platform-specific.
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/rust-analyzer/rust-analyzer/issues/5918#issuecomment-769214242

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by lnicola** [comment]
Now I get an error building `alsa-sys` (via `cpal`) -- I don't think ALSA is available on OpenBSD?

**Comment by AaronM04** [comment]
Correct. If you switch to the aaron/openbsd_ggez branch, you should be able to test this, since it uses sndio rather than ALSA. Note that this branch has submodules.

Sorry this is more complex than you probably anticipated. I think you're almost there :)

On January 31, 2021 2:17:24 AM PST, "Laurențiu Nicola" <notifications@github.com> wrote:
>Now I get an error building `alsa-sys` (via `cpal`) -- I don't think
>ALSA is available on OpenBSD?
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/rust-analyzer/rust-analyzer/issues/5918#issuecomment-770358769

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by lnicola** [comment]
Looks like it also needs:

```
# pkg_add clang-tools-extra wireshark
$ export LIBCLANG_PATH=/usr/local/lib/
```

But `dissect-netwayste` still doesn't build:

```
  wrapper.h:2:10: fatal error: 'config.h' file not found
  wrapper.h:2:10: fatal error: 'config.h' file not found, err: true
  thread 'main' panicked at 'Unable to generate bindings: ()', dissect-netwayste/build.rs:28:48
```

So I excluded it again from the workspace. I then installed `rust-analyzer` at `1a59f75` and ran:

```
$ export PATH=~/.cargo/bin:$PATH
```

I set up `nvim` using https://github.com/lnicola/dotfiles/blob/master/.config/nvim/init.vim, but disabled `deoplete`:

```vimscript
"Pack 'Shougo/deoplete.nvim', { 'do': ':UpdateRemotePlugins' }
```

And built `LanguageClient-neovim` using:

```
$ cd ~/.config/nvim/pack/minpac/start/LanguageClient-neovim
$ ./install.sh
```

Then tried to load the project. However, OpenBSD doesn't seem to package `rust-src`. so I copied over my installation from my Linux box and ran:

```
$ export RUST_SRC_PATH=~/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library
```

Opened `conwayste/src/client.rs`, added that line and got:

![image](https://user-images.githubusercontent.com/308347/106457778-21372200-6498-11eb-8d91-eeaf2828fc03.png)

One issue is that indeed, the error seems to be reported as a warning. It's an error in the beginning (there's an `x` in the left`), but it disappears after a couple of seconds. It does, however, say error in the status bar:

![image](https://user-images.githubusercontent.com/308347/106457885-4fb4fd00-6498-11eb-9f95-758ea2bd8449.png)

Do the other `rust-analyzer` features work for you? How did you acquire the standard library source code?

**Comment by lnicola** [comment]
@AaronM04 are you still having this issue on code that compiles? I guess I can send you my OpenBSD VM image if you want to give it a try.

**Comment by AaronM04** [comment]
Hi @lnicola, sorry for the late reply.

Yes, I am still having this issue, now reproduced with:
```
$ rust-analyzer --version
rust-analyzer 3ab8d7a
$ rustc --version
rustc 1.49.0
$ cd conwayste/
$ git rev-parse --short HEAD  # this commit compiles
b45f848
```

I am able to use other rust-analyzer features, such as going to a definition. Clearly the analyzer has some understanding of the semantics of the code. For some reason, though, it doesn't report this error.

If it's not too much trouble, can you send me your VM image?

**Comment by AaronM04** [comment]
I was able to reproduce this with a more recent rust-analyzer commit:

```
rust-analyzer 14de9e5
```

**Comment by AaronM04** [comment]
For the first time, I was _not_ able to reproduce this, a few days ago. I don't know why.

**Comment by lnicola** [comment]
@AaronM04 sorry for the late answer, I realized I probably had to make another VM and it got on the back burner. Is it working for you now?

**Comment by lnicola** [comment]
Glad to hear it worked out in the end. I'll close this, but feel free to reopen if you run into this again.

**Comment by danakj** [comment]
I've had this problem occur for me today. The analyzer can syntax highlight and find syntax errors, but cargo check output is ignored. I narrowed it down starting when I wrote a .cargo/config.toml file in my project root.

Having things in the project's .cargo/config.toml does not cause problems in general. And by changing the config.toml and forcing cargo check to run again, it seems to have cleared up whatever was broken. Reisntalling/upgrading/downgrading the analyzer did not.

So, in the end it's working for me again, and I don't know what broke it. Perhaps cargo check failed mid-way through for some reason and left things in a bad state that could not be recovered from without dirtying everything (changing profile.dev.codegen-units for example).
